json_merger/dict_merger.py

```diff
@@ -139,14 +139,14 @@ def conflicts(self):
     def _merge_base_values(self):
         if self.head == self.update:
             self.merged_root = self.head
-        elif self.head == NOTHING:
-            self.merged_root = self.update
         elif self.update == NOTHING:
             self.merged_root = self.head
         elif self.head == self.root:
             self.merged_root = self.update
         elif self.update == self.root:
             self.merged_root = self.head
+        elif self.head == NOTHING:
+            self.merged_root = self.update
```

Comment:
why is this still needed? isn't this entirely taken care of by the additional check you're doing when checking for `NOTHING` later on?

Reply:
There are some cases when HEAD is None and update isn't same as the root, so I thought that we want to take update in that case.

Reply:
ok, that makes sense.